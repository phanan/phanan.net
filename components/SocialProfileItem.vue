<template>
  <li>
    <a :href="profile.url">{{ profile.name }}</a>
  </li>
</template>

<script>
import isLinkObject from '~/validators/isLinkObject'

export default {
  name: 'Social-Profile-Item',

  props: {
    profile: {
      type: Object,
      required: true,
      validator: isLinkObject
    }
  }
}
</script>

<style lang="scss" scoped>
@import "../assets/scss/_mixins";

li {
  display: inline-block;
  position: relative;
  margin-right: 5px;
  font-size: 20px;

  a {
    color: rgba(255, 255, 255, .5);

    &:hover {
      color: #fff;
    }
  }

  &::before {
    @include slash;
  }

  &:first-child::before {
    display: none;
  }

  @media only screen and (max-width: 640px) {
    font-size: 15px;
  }
}
</style>
