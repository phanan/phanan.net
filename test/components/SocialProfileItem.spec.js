import { renderToString } from '@vue/server-test-utils'
import SocialProfileItem from '~/components/SocialProfileItem'

describe('SocialProfileItem.vue', () => {
  it('matches snapshot', () => {
    expect(renderToString(SocialProfileItem, {
      propsData: {
        profile: {
          name: 'Foo Bar',
          url: 'https://foo.bar'
        }
      }
    })).toMatchSnapshot()
  })
})
