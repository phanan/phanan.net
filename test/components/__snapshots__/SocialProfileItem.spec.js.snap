// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialProfileItem.vue matches snapshot 1`] = `
<li>
  <a href="https://foo.bar">Foo Bar</a>
</li>
`;
